services/tables/src/main/java/com/linkedin/openhouse/tables/utils/TableUUIDGenerator.java

```diff
@@ -261,7 +264,17 @@ private UUID extractUUIDFromExistingManifestListPath(
    */
   private boolean doesPathExist(java.nio.file.Path tableDirPath) {
     try {
-      return fsStorageProvider.storageClient().exists(new Path(tableDirPath.toString()));
+      // TODO: Refactor client interaction to use high-level Storage API such as
+      // StorageManager::doesObjectExist
+      if (storageManager.getDefaultStorage().getType().equals(StorageType.HDFS)
+          || storageManager.getDefaultStorage().getType().equals(StorageType.LOCAL)) {
+        FileSystem fs =
+            (FileSystem) storageManager.getDefaultStorage().getClient().getNativeClient();
+        return fs.exists(new Path(tableDirPath.toString()));
+      } else {
+        throw new UnsupportedOperationException(
```

Comment:
similar ask as secure block. This one is even more critical. 
But glad to see  comments "Refactor client interaction to use high-level Storage API such as
       // StorageManager::doesObjectExist"

In my opinion this should be prioritized.

Reply:
Yes! this is relatively high priority and will be addressed separately. Impact of not solving this sooner is new storages won't support CTAS/Replication.

btw, Do you like the approach of `StorageManager::doesObjectExist`?

Reply:
Got it! we should see some PRs soon! Quoting other thread:
>I want any high-level expectations of OH to the storage to be in the Storage class, example could be: Storage::allocateTableObjectSpace , Storage::doesTableObjectExist